src/simulation/LoadGenerator.cpp

```diff
@@ -1646,21 +1635,20 @@ LoadGenerator::invokeSorobanCreateUpgradeTransaction(
 std::pair<LoadGenerator::TestAccountPtr, TransactionFramePtr>
 LoadGenerator::sorobanRandomWasmTransaction(uint32_t ledgerNum,
                                             uint64_t accountId,
-                                            SorobanResources resources,
-                                            size_t wasmSize,
                                             uint32_t inclusionFee)
 {
+    auto [resources, wasmSize] = sorobanRandomUploadResources();
+
     auto account = findAccount(accountId, ledgerNum);
-    Operation uploadOp =
-        createUploadWasmOperation(static_cast<uint32>(wasmSize));
+    Operation uploadOp = createUploadWasmOperation(wasmSize);
     LedgerKey contractCodeLedgerKey;
     contractCodeLedgerKey.type(CONTRACT_CODE);
     contractCodeLedgerKey.contractCode().hash =
         sha256(uploadOp.body.invokeHostFunctionOp().hostFunction.wasm());
     resources.footprint.readWrite.push_back(contractCodeLedgerKey);
 
-    int64_t resourceFee =
-        sorobanResourceFee(mApp, resources, 5000 + wasmSize, 100);
+    int64_t resourceFee = sorobanResourceFee(
+        mApp, resources, 5000 + static_cast<size_t>(wasmSize), 100);
     // Roughly cover the rent fee.
     resourceFee += 100000;
```

Comment:
I'm hesitant to change this because it's working fine with a variety of distribution settings at the moment. It may require adjusting in the future, but that's true of all of the estimates in loadgen.

Reply:
Ya that's fine, I think it should be significantly high, especially since the BucketList size is very small for loadgen.